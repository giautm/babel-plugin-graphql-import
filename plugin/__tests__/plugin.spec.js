import BabelGraphQLImportPlugin from '../index';
import * as babel from 'babel-core';

describe('Babel GraphQL Import - Plugin', () => {
  describe('Babel Plugin', () => {
    it('transforms the import statement into a variable with the intended content', () => {
      const transformedCode = babel.transform("import SomeExample from './fixtures/example.graphql';", {
        filename: __filename,
        plugins: [BabelGraphQLImportPlugin]
      });

      expect(transformedCode.code).toMatchSnapshot();
    });

    it('accepts different extensions', () => {
      const transformedCode = babel.transform("import SomeExample from './fixtures/example.gql-v2';", {
        filename: __filename,
        plugins: [[
          BabelGraphQLImportPlugin, {
            extensions: [
              '.gql-v2'
            ]
          }
        ]]
      });

      expect(transformedCode.code).toMatchSnapshot();
    });

    it('throws error when importing with destructuring', () => {
      expect(() => {
        babel.transform("import { SomeExample, AnotherExample } from './fixtures/example.graphql';", {
          filename: __filename,
          plugins: [BabelGraphQLImportPlugin]
        });
      }).toThrowError(Error);
    });
  });
});
