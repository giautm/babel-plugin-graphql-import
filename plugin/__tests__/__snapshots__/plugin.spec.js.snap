// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel GraphQL Import - Plugin Babel Plugin accepts different extensions 1`] = `"/* babel-plugin-graphql-import './fixtures/example.gql-v2' */var SomeExample = 'type Object {\\\\n  hello: String\\\\n}\\\\n';"`;

exports[`Babel GraphQL Import - Plugin Babel Plugin transforms the import statement into a variable with the intended content 1`] = `"/* babel-plugin-graphql-import './fixtures/example.graphql' */var SomeExample = 'type Object {\\\\n  hello: String\\\\n}\\\\n';"`;
